### ISSUE TITLE
FileLoadException out of the box running AspNet sample

### ISSUE THREAD

**Issue by Fram90** [open-telemetry/opentelemetry-dotnet#2239]
# Bug Report

List of [NuGet packages](https://www.nuget.org/profiles/OpenTelemetry) and
version that you are using:

Using examples/Examples.AspNet project, that references current source files.

In my project I use:
OpenTelemetry 1.1.0
OpenTelemetry.Api 1.1.0
OpenTelemetry.Instrumentation.Http 1.0.0-rc7
OpenTelemetry.Exporter.Console 1.1.0

*

Runtime version `net48`,`net461`

*

## Symptom

When I run an app with enabled tracing a FileLoadException occurs, referencing an `System.Diagnostics.DiagnosticSource` assembly.
Removing `<add name="TelemetryCorrelationHttpModule" type="Microsoft.AspNet.TelemetryCorrelation.TelemetryCorrelationHttpModule, Microsoft.AspNet.TelemetryCorrelation" preCondition="integratedMode,managedHandler"/>` helps to run the app, in case AspNet, but it is not an option. 
In case of ConsoleApp FileNotFoundException references `System.Runtime.CompilerServices.Unsafe` assembly.

Sorry for local error text, but I think you've got the idea.
![image](https://user-images.githubusercontent.com/11601465/128531873-4359d352-561f-440c-aaf9-47c56a99a6fa.png)

**What is the expected behavior?**

I expect application to run and collect traces.

**What is the actual behavior?**

App doesn't run at all.

## Reproduce

The repo,  examples/Examples.AspNet

## Additional Context

It seems that OpenTelemetry uses DiagnoticSource v5.0.1, and something else in basic packages uses another version. Somehow these two versions conflict with each other. I've tried handle this with `<bindingRedirect>`, but had no luck.

**Comment by kyle-miller-cko** [comment]
Having same issue, did you find a fix?

**Comment by Fram90** [comment]
@kyle-miller-cko nope. Fortunately I use Elastic APM to store traces, so I switched to Elastic.APM Agent to collect traces from my app. Works without any issues on both classic .Net Framework and .Net Core.
